python/ray/tests/test_reference_counting_2.py

```diff
@@ -645,6 +645,29 @@ def test(self):
         time.sleep(1)
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="Failing on Windows.")
+def test_deep_nested_refs(shutdown_only):
+    ray.init(object_store_memory=100 * 1024 * 1024)
+
+    @ray.remote
+    def f(x):
+        print(f"=> step {x}")
+        if x > 200:
+            return x
+        return f.remote(x + 1)
+
+    r = f.remote(1)
+    i = 0
+    while isinstance(r, ray.ObjectRef):
+        print(i, r)
+        i += 1
+        try:
+            r = ray.get(r)
+        except ray.exceptions.ReferenceCountingAssertionError:
```

Comment:
Should remove this catch now right? I don't see it happening when I run with the patch.

Reply:
Oh nevermind, I see the other issue...

Reply:
Yeah, this PR just fixes the assertion check/cluster crash. The other issue has probably been in the codebase since ref counting was introduced...